sdk/core/src/services/WalletStorageService.ts

```diff
@@ -293,93 +94,100 @@ export default class WalletStorageService {
-      encryptedCredentials.push(cyphertext)
-    }
+    const credentials = await this._affinidiVaultStorageService.getAllCredentials(types, storageRegion)
+
+    // should be deleted during migration to affinidi-vault Phase #2
+    const bloomCredentials = await this._bloomVaultStorageService.getAllCredentials(types, storageRegion)
 
-    return await this.saveCredentials(encryptedCredentials, storageRegion)
+    return [...credentials, ...bloomCredentials]
   }
 
-  async fetchAllDecryptedCredentials() {
-    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
-    const allBlobs = await this.fetchAllBlobs()
-    const allCredentials = []
-    for (const blob of allBlobs) {
-      const credential = await this._platformEncryptionTools.decryptByPrivateKey(privateKeyBuffer, blob.cyphertext)
-      allCredentials.push(credential)
+  public async getCredentialsByShareToken(token: string, storageRegion?: string) {
+    storageRegion = storageRegion || this._storageRegion
+
+    if (!token) {
+      return this.getAllCredentials()
     }
 
-    return allCredentials
-  }
+    const request = JwtService.fromJWT(token)
+    const {
+      payload: {
+        interactionToken: { credentialRequirements },
+      },
+    } = request
 
-  async fetchDecryptedCredentials(fetchCredentialsPaginationOptions: FetchCredentialsPaginationOptions) {
-    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
-    const blobs = await this.fetchEncryptedCredentials(fetchCredentialsPaginationOptions)
-    const credentials = []
-    for (const blob of blobs) {
-      const credential = await this._platformEncryptionTools.decryptByPrivateKey(privateKeyBuffer, blob.cyphertext)
-      credentials.push(credential)
-    }
+    const requirementTypes = credentialRequirements.map(
+      (credentialRequirement: { type: string[] }) => credentialRequirement.type,
+    )
 
-    return credentials
+    return this.getAllCredentials(requirementTypes, storageRegion)
   }
 
-  async getCredentialByIndex(credentialIndex: number): Promise<any> {
-    const paginationOptions: FetchCredentialsPaginationOptions = { skip: credentialIndex, limit: 1 }
-    const credentials = await this.fetchDecryptedCredentials(paginationOptions)
+  public async getCredentialById(credentialId: string, storageRegion?: string): Promise<any> {
+    storageRegion = storageRegion || this._storageRegion
 
-    if (!credentials[0]) {
-      throw new SdkErrorFromCode('COR-14')
-    }
+    try {
+      const encryptedCredential = await this._affinidiVaultStorageService.getCredentialById(credentialId, storageRegion)
+      return encryptedCredential
+    } catch (error) {
+      if (error.code !== 'AVT-2') {
+        throw error
+      }
 
-    return credentials[0]
+      // should be deleted during migration to affinidi-vault Phase #2
+      try {
+        return await this._bloomVaultStorageService.getCredentialById(credentialId, storageRegion)
+      } catch (bloomError) {
```

Comment:
No need to catch anything there.
(And if Bloom Vault returned some unexpected error, probably better if the caller receives it, instead of it being hidden by AVT-2)

Reply:
So, here I just was following the ADR, which says:

> delete operations: should go to both Bloom Vault and Affinidi Vault. Returning the first successful operation or error of Affinidi Vault in case both fail.

I thought Bloom vault is working as reserve, so main vault provider is Affinidi vault?

Reply:
Removed catch from Bloom Vault call